cypress/support/tooltip.po.ts

```diff
@@ -7,8 +7,112 @@ export class TooltipPo extends BaseComponent {
+      name: 'Tooltip on left'
+    },
+    placementBottom: {
+      tooltipContainerClass: 'tooltip in tooltip-bottom bs-tooltip-bottom bottom show',
+      name: 'Tooltip on bottom'
+    }
   };
+
+  isTooltipPlacementCorrect(baseSelector: string, placement: string, buttonName: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`)
+      .contains(buttonName)
+      .focus();
+
+    cy.get(`${baseSelector} ${this.containerTooltip}`)
+      .should('be.visible')
+      .should('have.attr', 'class', placement);
+  }
+
+  isTooltipAppearOnFocus(baseSelector: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`).focus();
+    this.isTooltipAppearsAndVisible(baseSelector);
+  }
+
+  isTooltipAppearOnClick(baseSelector: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`).click();
+    this.isTooltipAppearsAndVisible(baseSelector);
+  }
+
+ isTooltipAppearsAndVisible(baseSelector: string) {
+    cy.get(`${baseSelector}`).should('to.have.descendants', this.containerTooltip);
+    cy.get(`${baseSelector} ${this.containerTooltip}`).should('be.visible');
+  }
+
+  isTooltipDismissAfterTheClicks(baseSelector: string) {
+    cy.get(`${baseSelector}`).closest('.section.bd-example').click();
+    cy.get(`${baseSelector}`).should('not.to.have.descendants', this.containerTooltip);
+  }
+
+  isAppendToBodyTooltipAppearsVisibleAndDismissible(baseSelector: string, buttonName: string) {
+    cy.get(`${baseSelector}`).contains(buttonName).focus();
+    cy.get(`${this.containerTooltip}`).contains('Vivamus sagittis lacus vel augue laoreet rutrum faucibus');
+    this.isTooltipAppearsAndVisible(this.body);
+      }
+
+  isConfiguringTooltipAppearOnFocus(baseSelector: string) {
+    cy.get(`${baseSelector} ${this.togglerTooltip}`).focus();
+    this.isTooltipAppearsAndVisible(this.body);
+    }
+  isCustomTriggersClickAppearsAndVisible(baseSelector: string, buttonName: string, tiggerEvent: string, tooltipText: string) {
+    if (tiggerEvent === 'click') {
+    cy.get(`${baseSelector}`).contains(buttonName).click();
+    } else {
+      cy.get(`${baseSelector}`).contains(buttonName).trigger(tiggerEvent);
+    }
+    cy.get(`${baseSelector}`).should('to.have.descendants', this.containerTooltip);
+    cy.get(`${this.containerTooltip}`).should('be.visible').contains(tooltipText);
+    cy.get(`${baseSelector}`).contains(buttonName).click();
+    cy.get(`${baseSelector}`).should('not.to.have.descendants', this.containerTooltip);
+  }
+
+  isManualTooltipsAppearsAndVisible(baseSelector: string, tooltipText: string) {
+    cy.get(`${baseSelector}`).contains('Show').click();
+    cy.get(`${baseSelector}`).should('to.have.descendants', this.containerTooltip);
+    cy.get(`${this.containerTooltip}`).should('be.visible').contains(tooltipText);
+    cy.get(`${baseSelector}`).contains('Hide').click();
+    cy.get(`${baseSelector}`).should('not.to.have.descendants', this.containerTooltip);
```

Comment:
all such actions + assertions should not be in 1 method. For click on buttons use method from base component